### ISSUE TITLE
CocoaPods installs dependency as static libraries after upgrading to 1.0.0

### ISSUE THREAD

**Issue by ngocluu** [CocoaPods/CocoaPods#5322]
# Report

## What did you do?

I integrate SVProgressHUD for main app and app extension.

1. Upgrade CocoaPods to 1.0.0 from 0.39.0.
2. Migrate podfile to 1.0.0 spec.
3. Run `pod install`.

## What did you expected to happen?

Target for app extension requires macro `SV_APP_EXTENSIONS`, CocoaPods would install SVProgressHUD as two frameworks: one for the main app (no macro) and one for the app extension.

## What happened instead?

CocoaPods 0.39.0 installs SVProgressHUD as two frameworks. But CocoaPods 1.0.0 installs SVProgressHUD as two static libraries.

## Podfile

```
use_frameworks!

abstract_target 'MyExample' do
    pod 'SVProgressHUD', '1.1.3'
    # other pods
    
    # Main app

    target 'MyExampleMain' do
        # other pods
    end

    # Share extension

    target :'MyExampleShareExtension' do
        # other pods
    end
end

pre_install do |installer|
    pod_targets = installer.pod_targets.flat_map do |pod_target|
        pod_target.name == "SVProgressHUD" ? pod_target.scoped : pod_target
    end
    installer.aggregate_targets.each do |aggregate_target|
        aggregate_target.pod_targets = pod_targets.select do |pod_target|
            pod_target.target_definitions.include?(aggregate_target.target_definition)
        end
    end
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'SVProgressHUD-Pods-MyExample-MyExampleShareExtension'
      target.build_configurations.each do |config|
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)']
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'SV_APP_EXTENSIONS'
      end
    end
  end
end
```

**Comment by neonichu** [comment]
The cause of this seems to be your custom `pre_install` hook. Not sure what you're trying to do with it.

**Comment by ngocluu** [comment]
This was fixed in Cocoapods 1.2.0. Thanks!
